test/malli/roundtrip_test.clj

```diff
@@ -62,4 +68,227 @@
     (is (nil? (rt/explain-roundtrip
                 [:orn [:i [:int]] [:s [:string]]])))
     (is (nil? (rt/explain-roundtrip
-                [:orn [:a [:any]] [:b [:nil]]])))))
+                [:orn [:a [:any]] [:b [:nil]]]))))
+  (testing ":orn with overlapping types is still roundtrippable"
+    ;; Even though int and number overlap, :orn makes it roundtrippable
+    (is (nil? (rt/explain-roundtrip
+                [:orn [:i [:int]] [:n number?]])))))
+
+(deftest roundtrippable-map
+  (testing ":map with roundtrippable children is roundtrippable"
+    (is (nil? (rt/explain-roundtrip
+                [:map [:x [:int]] [:y [:string]]]))))
+  (testing ":map with non-roundtrippable children is not roundtrippable"
+    (let [result (rt/explain-roundtrip
+                   [:map [:x [:or [:orn [:i [:int]]] [:orn [:j [:int]]]]]])]
+      (is (vector? result))
+      (is (seq result)))))
```

Comment:
For every non-roundtripping test, include a real call to demonstrate as well as document the actual return, of the form `(is (= v2 (m/unparse s (m/parse s v1))))`.